code/iaas/storage-service/src/main/java/io/cattle/platform/storage/api/filter/ExternalTemplateInstanceFilter.java

```diff
@@ -55,14 +60,34 @@ public Object create(String type, ApiRequest request, ResourceManager next) {
         return super.create(type, request, next);
     }
 
-    protected Image validateImageUuid(SchemaFactory schemaFactory, String uuid, Long registryCredentialId) {
+    protected Image validateImageUuid(SchemaFactory schemaFactory, String uuid, Instance instance) {
         try {
             Image image = storageService.registerRemoteImage(uuid);
             if (image == null) {
                 return null;
             }
-            if (registryCredentialId != null) {
-                objectManager.setFields(image, IMAGE.REGISTRY_CREDENTIAL_ID, registryCredentialId);
+            Long id = instance.getRegistryCredentialId();
+            if (id == null && image.getFormat().equalsIgnoreCase(DockerStoragePoolConstants.DOCKER_FORMAT)) {
+                String type = schemaFactory.getSchemaName(StoragePool.class);
+                List<?> storagePools = locator.getResourceManagerByType(type).list(type, new HashMap<>(), new ListOptions());
```

Comment:
This question is a bit for my own understanding.  Was there a particular reason we chose to retrieve a list of storagePools and credentials using the ResourceManager instead of ObjectManager?

I haven't dug into the code deeper, but it looks like the ResourceManager is closer to the public API (hence, the ListOptions for pagination, etc...) so it could potentially be performing more checks/filtering than ObjectManager which probably provides a more low level API.